crates/sui/src/sui_move/prove.rs

```diff
@@ -30,16 +32,33 @@ impl Prove {
             args.push(format!("--config={}", prover_toml.to_string_lossy()));
         }
         args.extend(opts.iter().cloned());
-        let options = move_prover::cli::Options::create_from_args(&args)?;
 
-        prove::run_move_prover(
-            build_config,
-            &rerooted_path,
-            &target_filter,
-            for_test,
-            options,
-        )?;
+        let mut options = move_prover::cli::Options::create_from_args(&args)?;
+
+        // provide Sui-specific Boogie template for the native functions to the prover.
+        options.backend.custom_natives =
+            Some(move_prover_boogie_backend::options::CustomNativeOptions {
+                template_bytes: SUI_NATIVE_TEMPLATE.to_vec(),
+                module_instance_names: vec![
+                    (
+                        "0x2::transfer".to_string(),
+                        "transfer_instances".to_string(),
+                    ),
+                    ("0x2::id".to_string(), "id_instances".to_string()),
+                    ("0x2::event".to_string(), "sui_event_instances".to_string()),
+                ],
+            });
+
+        let prover_result = std::thread::spawn(move || {
+            prove::run_move_prover(
+                build_config,
+                &rerooted_path,
+                &target_filter,
+                for_test,
+                options,
+            )
+        });
 
-        Ok(())
+        prover_result.join().expect("move prover thread panicked")
```

Comment:
would `prover_result.join()?` work here? Or, if the error type is different than `anyhow`, perhaps we can do some conversion to simply propagate it rather than using `except`?

Reply:
Good point. `prover_result.join()?` doesn't work here since the error type is different. I'll add some conversion.

Reply:
Hmm I tried doing `prover_result.join().unwrap_or_else(|err| { Err(anyhow::Error::new(err)) })` hoping to convert the `err` but it doesn't compile, saying "the size for values of type `dyn std::any::Any + Send` cannot be known at compilation time". Rust result/error is confusing. Do you have a better idea of how this should work? @awelc 

Btw IIUC when we use `expect` the errors generated by the prover will be propagated. The error that will get unwrapped away is if`join`ing the thread fails.

Reply:
I am not sure I understand... I believe that `expect` will cause a panic if executed. How about something like this:
```
        prover_result
            .join()
            .map_or_else(|err| Err(anyhow::anyhow!("{:?}", err)), |_| Ok(()))
```
Either way it's a nit so feel free to merge if you don't like it.

Reply:
Ah yes using `anyhow!` would work. Thanks!